test/getJwk.spec.js

```diff
@@ -0,0 +1,158 @@
+'use strict'
+
+const nock = require('nock')
+const t = require('tap')
+
+const buildGetJwks = require('../src/get-jwks')
+
+t.beforeEach((done) => {
+  nock.disableNetConnect()
+  done()
+})
+
+t.afterEach((done) => {
+  nock.cleanAll()
+  nock.enableNetConnect()
+  done()
+})
+
+t.test('getJwk should return an error if the request fails', async t => {
+  nock('https://localhost/').get('/.well-known/jwks.json').reply(500, { msg: 'no good' })
+  try {
+    const getJwks = buildGetJwks()
+    await getJwks.getJwk({ domain: 'https://localhost/', alg: 'ALG', kid: 'SOME_KEY' })
+  } catch (e) {
+    t.equal(e.message, 'Internal Server Error')
+    t.same(e.body, { msg: 'no good' })
+  }
+  t.throws(() => { throw new Error('An error was thrown') }, {})
```

Comment:
useless

Reply:
Thanks for showing that, I see how completely useless that is now!